pkg/odo/cli/deploy/deploy.go

```diff
@@ -105,28 +106,36 @@ func (o *DeployOptions) Complete(cmdline cmdline.Cmdline, args []string) (err er
 		return err
 	}
 
-	envFileInfo, err := envinfo.NewEnvSpecificInfo(o.contextDir)
+	// ENV.YAML
+	//
+	// Set the appropriate variables in the env.yaml file
+	//
+	// TODO: Eventually refactor with code in dev/dev.go
+
+	envfileinfo, err := envinfo.NewEnvSpecificInfo("")
 	if err != nil {
 		return fmt.Errorf("unable to retrieve configuration information: %w", err)
 	}
-	if !envFileInfo.Exists() {
-		var cmpName string
-		cmpName, err = component.GatherName(o.EnvSpecificInfo.GetDevfileObj(), o.GetDevfilePath())
-		if err != nil {
-			return fmt.Errorf("unable to retrieve component name: %w", err)
-		}
-		err = envFileInfo.SetComponentSettings(envinfo.ComponentSettings{Name: cmpName, Project: o.GetProject(), AppName: "app"})
+
+	// If the env.yaml does not exist, we will save the project name
+	if !envfileinfo.Exists() {
+		err = envfileinfo.SetComponentSettings(envinfo.ComponentSettings{Project: o.GetProject()})
 		if err != nil {
 			return fmt.Errorf("failed to write new env.yaml file: %w", err)
 		}
+	}
 
-	} else if envFileInfo.GetComponentSettings().Project != o.GetProject() {
-		err = envFileInfo.SetConfiguration("project", o.GetProject())
+	// If the env.yaml exists and the project is set incorrectly, we'll override it.
+	if envfileinfo.Exists() && envfileinfo.GetComponentSettings().Project != o.GetProject() {
```

Comment:
nit: The previous `else if envfileinfo.GetComponentSettings().Project != o.GetProject()` makes the condition clearer, I think.

Reply:
Yes, but we also need envfileinfo.Exists() for nil check as well, so that's why it's added

Reply:
Not sure I understand.. Which nil check are you referring to? 
With the `else if` statement, we are sure that `envfileinfo.Exists()` evaluates to `true`, so we don't need to check it again.
I just find the code to be more readable with the `else if` that was there before.